src/api/app/controllers/webui/request_controller.rb

```diff
@@ -292,6 +292,18 @@ def conversation
     @active = 'conversation'
   end
 
+  def build_results
+    redirect_to request_show_path(params[:number], params[:request_action_id]) unless @action[:sprj] || @action[:spkg]
+
+    @active = 'build_results'
+    @project = @staging_project || @action[:sprj]
+    @buildable = @action[:spkg] || @project
+
+    @ajax_data = {}
+    @ajax_data['project'] = @project if @project
+    @ajax_data['package'] = @action[:spkg] if @action[:spkg]
```

Comment:
Can this be squeezed into something like

```
@ajax_data = {
  project: @project if @project,
  package: @action[:spkg] if @action[:spkg]
}
```

Could you also explain why do you check `if @project` and `if @action[:spkg]` in the assignment? I mean, if the condition would be false it means there is no `@project`, but there is no `else` so I don't really see the difference to assign directly the parameter no matter what the value is. It would result into something like:
```
@ajax_data = {
  project: @project,
  package: @action[:spkg]
}
```
Am I missing something?
If the expectation is something different maybe you want to use a ternary operator and assign something else, just in case?

Reply:
The difference is that using the if will create the `project` key in `@ajax_data` only when there is something to fill the key with, otherwise you'll get a hash like `{ 'project': nil }`

Reply:
These values are used in initializing some HTML data attributes. I guess the checks are important.

Reply:
I prefer the collapsed way actually, but at this point it is more about a personal style feeling so, as long as the majority prefer this way, I am totally fine. Feel free to keep it as it is. Thanks for double checking btw. :+1: